src/rendergraph/common/rendergraph/material/endoftrackmaterial.cpp

```diff
@@ -0,0 +1,39 @@
+#include "endoftrackmaterial.h"
+
+#include <QVector2D>
+
+#include "rendergraph/materialshader.h"
+#include "rendergraph/materialtype.h"
+#include "rendergraph/uniformset.h"
+
+using namespace rendergraph;
+
+EndOfTrackMaterial::EndOfTrackMaterial()
+        : Material(uniforms()) {
+}
+
+/* static */ const AttributeSet& EndOfTrackMaterial::attributes() {
+    static AttributeSet set = makeAttributeSet<QVector2D, float>({"position", "gradient"});
```

Comment:
it be really nice if this could be made `constexpr static`. All the info is there so I'm sure it can be done if the underlying API was adjusted for it.

Reply:
I would appreciate it very much if you can come with something!

Reply:
I am trying something...

Reply:
No, I am getting stuck. I you can think of an elegant way to create an AttributeSet the describes a given list of types with named, and ideally check that this list of types matches a particular struct, that would be more welcome.

Reply:
I'm not quite sure I can follow you. If you're asking how I you can transform a list of kv pairs into a struct, thats gonna be difficult. There are a couple static reflection libraries that are capable of that but they're quite sophisticated.